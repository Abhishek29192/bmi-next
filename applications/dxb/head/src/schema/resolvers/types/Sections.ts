import type { ContentfulCardCollection } from "./CardCollection";
import type { ContentfulIframeSectionData } from "./IframeSection";
import type { ContentfulCarouselSection } from "./CarouselSection";
import type { Data as EmbeddedScriptSectionData } from "../../../components/EmbeddedScriptSection";
import type { ContentfulLeadBlock } from "./LeadBlock";
import type { ContentfulPromo } from "./Promo";
import type { ContentfulSignUpBlockData } from "./SignUpBlock";
import type { ContentfulTitleWithContentData } from "./TitleWithContent";
import type { ContentfulVideoSectionData } from "./VideoSection";
import type { ContentfulSyndicateSectionData } from "./SyndicateSection";
import type { ContentfulServiceLocatorData } from "./ServiceLocatorSection";

export type ContentfulSection =
  | ContentfulCardCollection
  | ContentfulCarouselSection
  | ContentfulIframeSectionData
  | ContentfulPromo
  | ContentfulServiceLocatorData
  | ContentfulSignUpBlockData
  | ContentfulSyndicateSectionData
  | ContentfulTitleWithContentData
  | ContentfulVideoSectionData
  | EmbeddedScriptSectionData
  | ContentfulLeadBlock;

export type ContentfulSections = ContentfulSection[];
