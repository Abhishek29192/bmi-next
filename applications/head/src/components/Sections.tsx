import React from "react";
import { graphql } from "gatsby";
import TabsOrAccordionSection, {
  Data as TabsOrAccordionSectionData
} from "../components/TabsOrAccordionSection";
import VillainSection, {
  Data as VillainSectionData
} from "../components/VillainSection";
import TwoPaneCarouselSection, {
  Data as TwoPaneCarouselSectionData
} from "../components/TwoPaneCarouselSection";

export type Data = (
  | TabsOrAccordionSectionData
  | VillainSectionData
  | TwoPaneCarouselSectionData
)[];

const sectionsMap = {
  ContentfulTabsOrAccordionSection: TabsOrAccordionSection,
  ContentfulVillainSection: VillainSection,
  ContentfulTwoPaneCarouselSection: TwoPaneCarouselSection
};

const Sections = ({ data }: { data: Data }) => {
  return (
    <>
      {data.map((section, index) => {
        const Component = sectionsMap[section.__typename];
        return (
          <Component
            key={`section${index}`}
            // @ts-ignore
            data={section}
            // TODO: Robust theme-based solution required.
            backgroundColor={index % 2 === 0 ? "pearl" : "white"}
          />
        );
      })}
    </>
  );
};

export default Sections;

export const query = graphql`
  # NOTE: This union type name is not ideal, but the best option so far.
  fragment SectionsFragment on ContentfulTabsOrAccordionSectionContentfulVillainSectionUnion {
    __typename
    ...TabsOrAccordionSectionFragment
    ...VillainSectionFragment
    # ...TwoPaneCarouselSectionFragment
  }
`;
