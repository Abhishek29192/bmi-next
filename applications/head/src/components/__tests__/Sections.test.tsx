import React from "react";
import Sections, { Data } from "../Sections";
import { render } from "@testing-library/react";
import { BLOCKS } from "@contentful/rich-text-types";
import mockConsole from "jest-mock-console";

beforeAll(() => {
  mockConsole();
});

describe("Sections component", () => {
  it("renders correctly", () => {
    const data: Data = [
      {
        __typename: "ContentfulTabsOrAccordionSection",
        description: { description: "string" },
        items: [
          {
            title: "hello",
            content: {
              json: {
                data: {},
                nodeType: BLOCKS.DOCUMENT,
                content: []
              }
            }
          }
        ],
        title: "string",
        type: "Accordion"
      },
      {
        __typename: "ContentfulTabsOrAccordionSection",
        description: { description: "string" },
        items: [
          {
            title: "hello",
            content: {
              json: {
                data: {},
                nodeType: BLOCKS.DOCUMENT,
                content: []
              }
            }
          }
        ],
        title: "string",
        type: "Tabs"
      },
      {
        __typename: "ContentfulVillainSection",
        title: "Title",
        promo: {
          __typename: "ContentfulPromo",
          title: "Promo",
          brandLogo: null,
          subtitle: null,
          featuredImage: null,
          cta: null
        }
      },
      {
        __typename: "ContentfulLeadBlockSection",
        text: {
          json: {
            data: {},
            content: [
              {
                data: {},
                content: [
                  {
                    data: {},
                    marks: [],
                    value: "lead block heading",
                    nodeType: "text"
                  }
                ],
                nodeType: BLOCKS.HEADING_2
              },
              {
                data: {},
                content: [
                  {
                    data: {},
                    marks: [],
                    value: "lead block content",
                    nodeType: "text"
                  }
                ],
                nodeType: BLOCKS.PARAGRAPH
              }
            ],
            nodeType: BLOCKS.DOCUMENT
          }
        },
        link: {
          __typename: "ContentfulLink",
          id: "00000000-0000-0000-0000-000000000000",
          label: "lead block cta label",
          icon: null,
          isLabelHidden: null,
          url: null,
          linkedPage: {
            slug: "contact-us"
          }
        },
        cardTheme: "pearl",
        cardSections: [
          {
            id: "00000000-0000-0000-0000-000000000000",
            hasUnderline: true,
            title: "card section 1 title",
            description: {
              description: "card section 1 description"
            },
            link: {
              __typename: "ContentfulLink",
              id: "00000000-0000-0000-0000-000000000000",
              label: "card section 1 link label",
              icon: null,
              isLabelHidden: null,
              url: "/",
              linkedPage: null
            },
            linkType: null
          }
        ]
      },
      {
        __typename: "ContentfulCardCollectionSection",
        title: "card collection section title",
        cardLabel: "Card Label",
        cardType: "Next Best Action Card",
        cards: [
          {
            __typename: "ContentfulPromo",
            title: "promo title",
            brandLogo: "AeroDek",
            subtitle: "promo subtitle",
            featuredImage: null,
            cta: {
              __typename: "ContentfulLink",
              id: "00000000-0000-0000-0000-000000000000",
              label: "Contact us",
              icon: null,
              isLabelHidden: null,
              url: null,
              linkedPage: {
                slug: "contact-us"
              }
            },
            id: "00000000-0000-0000-0000-000000000000"
          }
        ]
      },
      {
        __typename: "ContentfulCardCollectionSection",
        title: "card section title 2",
        cardLabel: "Go to {{title}}",
        cardType: "Product Overview Card",
        cards: [
          {
            __typename: "ContentfulSimplePage",
            id: "00000000-0000-0000-0000-000000000000",
            slug: "slug",
            title: "page title",
            brandLogo: null,
            subtitle: "page subtitle",
            featuredImage: {
              resize: {
                src: "//image-url"
              }
            }
          }
        ]
      }
    ];

    const { container } = render(<Sections data={data} />);
    expect(container.children).toMatchSnapshot();
  });
});
