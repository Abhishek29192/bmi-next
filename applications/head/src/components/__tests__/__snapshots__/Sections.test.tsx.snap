// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sections component renders correctly 1`] = `
HTMLCollection [
  <div
    id="string"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="Section Section--white"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl"
        >
          <h2
            class="MuiTypography-root title Typography Typography--underline MuiTypography-h2"
          >
            string
          </h2>
          <p
            class="MuiTypography-root Typography MuiTypography-body1"
          >
            string
          </p>
          <div
            style="margin-top: 40px;"
          >
            <div
              class="Accordion"
            >
              <div
                class="MuiPaper-root MuiAccordion-root item MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  aria-disabled="false"
                  aria-expanded="false"
                  class="MuiButtonBase-root MuiAccordionSummary-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiAccordionSummary-content"
                  >
                    <h3
                      class="MuiTypography-root Typography MuiTypography-h6"
                    >
                      hello
                    </h3>
                  </div>
                  <div
                    aria-disabled="false"
                    aria-hidden="true"
                    class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </div>
                <div
                  class="MuiCollapse-container MuiCollapse-hidden"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner"
                    >
                      <div
                        role="region"
                      >
                        <div
                          class="MuiAccordionDetails-root details"
                        >
                          <div
                            class="RichText"
                          >
                            <p
                              class="MuiTypography-root Typography MuiTypography-body1 MuiTypography-gutterBottom"
                            >
                              Lorem Ipsum has been the industry's standard dummy text ever since the 1500s
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="string"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="Section Section--pearl"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl"
        >
          <h2
            class="MuiTypography-root title Typography Typography--underline MuiTypography-h2"
          >
            string
          </h2>
          <p
            class="MuiTypography-root Typography MuiTypography-body1"
          >
            string
          </p>
          <div
            style="margin-top: 40px;"
          >
            <div
              class="Tabs Tabs--primary"
            >
              <div
                class="TabsBar"
              >
                <div
                  class="Container"
                >
                  <div
                    class="MuiContainer-root MuiContainer-maxWidthXl"
                  >
                    <div
                      class="MuiTabs-root"
                    >
                      <div
                        class="MuiTabs-scrollable"
                        style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                      />
                      <div
                        class="MuiTabs-scroller MuiTabs-scrollable"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          aria-label="tabs"
                          class="MuiTabs-flexContainer"
                          role="tablist"
                        >
                          <button
                            aria-controls="tabpanel-hello"
                            aria-selected="true"
                            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
                            id="tab-hello"
                            role="tab"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiTab-wrapper"
                            >
                              hello
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                        <span
                          class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorPrimary-2 MuiTabs-indicator"
                          style="left: 0px; width: 0px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                aria-labelledby="tab-hello"
                class="MuiGrid-root TabPanel MuiGrid-item"
                heading="hello"
                id="tabpanel-hello"
                role="tabpanel"
              >
                <div
                  class="MuiBox-root MuiBox-root-5 TabPanelBox"
                >
                  <div
                    class="RichText"
                  >
                    <p
                      class="MuiTypography-root Typography MuiTypography-body1 MuiTypography-gutterBottom"
                    >
                      Lorem Ipsum has been the industry's standard dummy text ever since the 1500s
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="Title"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="Section Section--white"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl"
        >
          <h2
            class="MuiTypography-root title Typography Typography--underline MuiTypography-h2"
          >
            Title
          </h2>
          <div
            class="Villain Villain--contained"
          >
            <div
              class="MuiGrid-root grid MuiGrid-container MuiGrid-spacing-xs-3"
            >
              <div
                class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
              >
                <div
                  class="content ColorPair ColorPair--blue-800"
                >
                  <h3
                    class="MuiTypography-root title Typography MuiTypography-h4"
                  >
                    Villain 1
                  </h3>
                  <div
                    class="text"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
              >
                <div
                  class="image"
                  style="background-image: url(undefined);"
                />
              </div>
            </div>
          </div>
          <div
            class="Villain Villain--contained Villain--reversed"
          >
            <div
              class="MuiGrid-root grid MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column-reverse"
            >
              <div
                class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
              >
                <div
                  class="content ColorPair ColorPair--blue-800"
                >
                  <h3
                    class="MuiTypography-root title Typography MuiTypography-h4"
                  >
                    Villain 2
                  </h3>
                  <div
                    class="text"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
              >
                <div
                  class="image"
                  style="background-image: url(undefined);"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="card collection section title"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="CardCollectionSection"
    >
      <div
        class="Section Section--white"
      >
        <div
          class="Container"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthXl"
          >
            <h2
              class="MuiTypography-root title Typography Typography--underline MuiTypography-h2"
            >
              card collection section title
            </h2>
            <div
              class="MuiGrid-root Grid MuiGrid-container MuiGrid-spacing-xs-3"
            >
              <div
                class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-xl-3"
              >
                <div
                  class="OverviewCard OverviewCard--flat"
                >
                  <div
                    class="body"
                  >
                    <svg
                      class="brand-logo"
                      preserveAspectRatio="xMinYMin"
                    />
                    <h4
                      class="MuiTypography-root title Typography Typography--underline MuiTypography-h4"
                    >
                      promo title
                    </h4>
                    <div
                      class="children"
                    >
                      promo subtitle
                    </div>
                    <div
                      class="footer"
                    >
                      <a
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Button MuiButton-outlinedPrimary"
                        href="/uk/contact-us"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <span
                            class="MuiButton-startIcon MuiButton-iconSizeMedium"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                              />
                            </svg>
                          </span>
                          Card Label
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-contained Button link MuiButton-containedPrimary"
              href="/uk/contact-us"
            >
              <span
                class="MuiButton-label"
              >
                Contact us
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="card section title 2"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="CardCollectionSection"
    >
      <div
        class="Section Section--pearl"
      >
        <div
          class="Container"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthXl"
          >
            <h2
              class="MuiTypography-root title Typography Typography--underline MuiTypography-h2"
            >
              card section title 2
            </h2>
            <div
              class="MuiGrid-root Grid MuiGrid-container MuiGrid-spacing-xs-3"
            >
              <div
                class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-xl-3"
              >
                <div
                  class="OverviewCard"
                >
                  <div
                    class="header-picture"
                    style="background-image: url(//image-url);"
                  />
                  <div
                    class="body"
                  >
                    <h4
                      class="MuiTypography-root title title--no-brand-logo Typography Typography--underline MuiTypography-h4"
                    >
                      page title
                    </h4>
                    <div
                      class="children"
                    >
                      page subtitle
                    </div>
                    <div
                      class="footer"
                    >
                      <a
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Button MuiButton-outlinedPrimary"
                        href="/uk/path"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <span
                            class="MuiButton-startIcon MuiButton-iconSizeMedium"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                              />
                            </svg>
                          </span>
                          Go to page title
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="card section title 2"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="PromoSection PromoSection Section Section--white"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl"
        >
          <div
            class="MuiGrid-root grid MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
            >
              <div
                class="content"
              >
                <h2
                  class="MuiTypography-root title Typography Typography--underline MuiTypography-h2"
                >
                  card section title 2
                </h2>
                <h3
                  class="MuiTypography-root subtitle Typography MuiTypography-h3"
                >
                  card section title 2
                </h3>
              </div>
            </div>
            <div
              class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
            >
              <div
                class="image"
                style="background-image: url(undefined);"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="Gallery title"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="ImageGallerySection Section Section--alabaster"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl"
        >
          <div
            class="MuiGrid-root Grid MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-8"
            >
              <h1
                class="MuiTypography-root Typography Typography--underline MuiTypography-h1"
              >
                Gallery title
              </h1>
              <p
                class="MuiTypography-root description Typography MuiTypography-body1"
              >
                card section 1 description
              </p>
            </div>
            <div
              class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="ImageGallery"
              >
                <div
                  class="main-image-wrapper main-image-wrapper--cover"
                  style="background-image: url(url);"
                >
                  <span
                    class="accessibility-label"
                  >
                    my test title
                  </span>
                </div>
                <div
                  class="thumbnails"
                >
                  <button
                    class="MuiButtonBase-root thumb-scroller thumb-scroller--left ThumbScrollerButton ThumbScrollerButton--left"
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root thumb-scroller thumb-scroller--right thumb-scroller--hidden ThumbScrollerButton ThumbScrollerButton--right"
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <div
                    class="scroller"
                    style="margin-right: 0%;"
                  >
                    <button
                      class="MuiButtonBase-root thumbnail Thumbnail Thumbnail--selected Thumbnail--large"
                      style="background-image: url(url);"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="accessibility-text"
                      >
                        my test title
                      </span>
                    </button>
                    <button
                      class="MuiButtonBase-root thumbnail Thumbnail Thumbnail--large"
                      style="background-image: url(url2);"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="accessibility-text"
                      >
                        my test title2
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="Document Downloads"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="Section Section--white"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl"
        >
          <h2
            class="MuiTypography-root title Typography Typography--underline MuiTypography-h2"
          >
            Document Downloads
          </h2>
          <div
            style="margin-bottom: 40px;"
          >
            <div
              class="RichText"
            >
              <p
                class="MuiTypography-root Typography MuiTypography-body1 MuiTypography-gutterBottom"
              >
                Lorem Ipsum has been the industry's standard dummy text ever since the 1500s
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="Service Locator"
    style="margin-top: -150px; padding-top: 150px;"
  >
    <div
      class="ServiceLocationSection Section Section--white"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl"
        >
          <div
            class="MuiGrid-root controls MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-flex-end"
          >
            <div
              class="MuiGrid-root search MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
            >
              <div
                aria-expanded="false"
                class="MuiAutocomplete-root company-autocomplete MuiAutocomplete-hasClearIcon"
                role="combobox"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root TextField MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                    data-shrink="true"
                    for="company-autocomplete"
                    id="company-autocomplete-label"
                  >
                    findARoofer.companyFieldLabel
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  >
                    <svg
                      class="start-adornment-icon"
                    />
                    <input
                      aria-autocomplete="list"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                      id="company-autocomplete"
                      name="autocomplete-field"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Clear"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                        tabindex="-1"
                        title="Clear"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
                      >
                        <span>
                          findARoofer.companyFieldLabel
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <p
                class="MuiTypography-root and-or-label Typography MuiTypography-body1"
              >
                <span>
                  findARoofer.andOr
                </span>
              </p>
              <div
                style="text-align: center;"
              >
                <div
                  class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text Button GeolocationButton MuiButton-textPrimary Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M21 3L3 10.53v.98l6.84 2.65L12.48 21h.98L21 3z"
                      />
                    </svg>
                  </span>
                  findARoofer.geolocationButton
                </span>
              </button>
            </div>
            <div
              class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-8"
            >
              <div
                class="filters"
              >
                <h6
                  class="MuiTypography-root Typography MuiTypography-h6"
                >
                  findARoofer.filtersLabel
                </h6>
                <div
                  class="chips"
                >
                  <div
                    class="MuiButtonBase-root MuiChip-root Chip Chip--theme-pearl MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-clickable"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiChip-label"
                    >
                      findARoofer.filters.pitchedRoof
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                  <div
                    class="MuiButtonBase-root MuiChip-root Chip Chip--theme-pearl MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-clickable"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiChip-label"
                    >
                      findARoofer.filters.flatRoof
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Tabs Tabs--secondary Tabs--visible-until-md tabs"
          >
            <div
              class="TabsBar"
            >
              <div
                class="Container"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthXl"
                >
                  <div
                    class="MuiTabs-root"
                  >
                    <div
                      class="MuiTabs-scroller MuiTabs-fixed"
                      style="overflow: hidden;"
                    >
                      <div
                        aria-label="tabs"
                        class="MuiTabs-flexContainer"
                        role="tablist"
                      >
                        <button
                          aria-controls="tabpanel-list"
                          aria-selected="true"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
                          id="tab-list"
                          role="tab"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiTab-wrapper"
                          >
                            findARoofer.listLabel
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="tabpanel-map"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
                          id="tab-map"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          <span
                            class="MuiTab-wrapper"
                          >
                            findARoofer.mapLabel
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <span
                        class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorPrimary-2 MuiTabs-indicator"
                        style="left: 0px; width: 0px;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-labelledby="tab-list"
              class="MuiGrid-root TabPanel tab-panel MuiGrid-item MuiGrid-grid-md-12 MuiGrid-grid-lg-4"
              heading="findARoofer.listLabel"
              id="tabpanel-list"
              role="tabpanel"
            >
              <div
                class="MuiBox-root MuiBox-root-10 TabPanelBox"
              >
                <div
                  class="list"
                >
                  <div
                    class="no-results"
                  >
                    <h4
                      class="MuiTypography-root no-results-heading Typography MuiTypography-h4"
                    >
                      findARoofer.noResults.title
                    </h4>
                    <p
                      class="MuiTypography-root Typography MuiTypography-body1"
                    >
                      findARoofer.noResults.subtitle
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-labelledby="tab-map"
              class="MuiGrid-root TabPanel tab-panel MuiGrid-item MuiGrid-grid-md-12 MuiGrid-grid-lg-8"
              heading="findARoofer.mapLabel"
              hidden=""
              id="tabpanel-map"
              role="tabpanel"
            >
              <div
                class="MuiBox-root MuiBox-root-11 TabPanelBox"
              >
                <div
                  class="map"
                >
                  <div
                    class="GoogleMap"
                  >
                    <div
                      class="map"
                    >
                      <div
                        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                        role="progressbar"
                        style="width: 40px; height: 40px;"
                      >
                        <svg
                          class="MuiCircularProgress-svg"
                          viewBox="22 22 44 44"
                        >
                          <circle
                            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                            cx="44"
                            cy="44"
                            fill="none"
                            r="20.2"
                            stroke-width="3.6"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
